### PR TITLE
90% Talis Ruby client should support assets being linked to multiple nodes

### PR DESCRIPTION
**Pull request by malcyL**
Talis Ruby client should support assets being linked to multiple nodes

### CODE REVIEW

File: lib/talis/hierarchy/asset.rb [hunk 1]
```diff
@@ -87,11 +84,13 @@ def initialize(namespace:, type:, id:, node: nil, attributes: {})
...
-                                                            @type,
-                                                            @id)
+        @nodes.each do |node|
+          self.class.api_client(request_id).add_asset_to_node(@namespace,
+                                                              node.type,
+                                                              node.id,
+                                                              @type,
+                                                              @id)
+        end
         mark_persisted
```
Comment (pads) [inline lib/talis/hierarchy/asset.rb]: I'm wondering how best we inform the caller that this is the case. The simplest thing to do would be to raise an exception saying that at least one node is required.

File: spec/integration/hierarchy/asset_spec.rb [hunk 2]
```diff
@@ -525,4 +559,11 @@ def node_bulk_upload(namespace, csv)
     request.use_ssl = (uri.scheme == 'https')
     expect(request.post(uri.path, csv, headers).code).to eq '204'
   end
+
+  def asset_associated_with_node(asset, namespace, type, id)
+    found_assets = Talis::Hierarchy::Asset.find_by_node(namespace: namespace,
+                                                        type: type,
+                                                        id: id)
+    expect(found_assets.any? { |a| a.id == asset.id }).to be true
```
Comment (pads) [inline spec/integration/hierarchy/asset_spec.rb]: I would be stricter and make sure the asset type matches too.

**Comment by pads** [comment]
Also, your commit messages appear to be stuck...

**Comment by malcyL** [comment]
Closing this pull request in favour of #25 

I had this crazy idea that we might want to accept these changes, but hold off on the save/update changes in #25. We#re not going to do that, so we're going to accept #25 not this PR.

**Comment by pads** [comment]
:+1:

**Pull request merged**